src/lava/magma/core/process/process.py

```diff
@@ -304,7 +304,7 @@ def is_sub_proc_of(self, proc: AbstractProcess):
             return False
 
     def run(self,
-            condition: AbstractRunCondition,
+            condition: ty.Optional[ty.Union[AbstractRunCondition, int]],
```

Comment:
Can we fix this inconsistency while we are at it? Why is one parameter called `condition` and the other `run_cfg`? Any of these combinations would make more sense to me:
- `config` and `condition` (prefer this)
- `run_config` and `run_condition` (okay)
- `run_cfg` and `run_cndtn` (ugh)

I also always get confused about which is which and look it up. Maybe there is a better naming entirely.

Reply:
I agree completely and have considered making this change several times, unfortunately this will be breaking for user code, so we'd need to do a deprecation thing and I put it off before.

I'll look into whether it's possible to deprecate just an argument (run_cfg) and add config.

Reply:
Do we really need to deprecate something at this stage? I feel we could start with that when reaching Lava version 1.0...